src/pages/questions/ProductInformation.jsx

```diff
@@ -38,10 +40,58 @@ const getImagesUrls = (images, barcode) => {
     .map((key) => `${rootImageUrl}/${key}.jpg`);
 };
 
+// src looks like: "https://static.openfoodfacts.org/images/products/004/900/053/2258/1.jpg"
+const getImageId = (src) => {
+  var sliced = src.split("/").at(-1);
+  sliced.replace(/\..+$/, "");
+  const imgid = sliced.slice(0, -4);
+  return Number(imgid);
+};
+
 const ProductInformation = ({ question }) => {
   const { t } = useTranslation();
   const [productData, setProductData] = React.useState({});
   const [hideImages, setHideImages] = React.useState(getHideImages);
+  const [flagged, setFlagged] = React.useState([]);
+
+  const flagImage = (src, barcode) => {
+    const imgid = getImageId(src);
+    try {
+      const response = axios.put(
+        `https://amathjourney.com/api/off-annotation/flag-image/${barcode}`,
+        {
+          mode: "no-cors",
+          imgid: imgid,
+          url: src,
+        }
+      );
+      const updatedImageArray = [...flagged];
+      updatedImageArray[imgid] = true;
+      setFlagged(updatedImageArray);
```

Comment:
I don't get what you're doing. You have an array of flagged ids. But how is it reset? For example I flag the second image. After answering the question I get a new product which should not be flagged

Maybe a useEffect to reset to an empty array would do the job

Reply:
mmm, what about not setting to `true` (it is not an `{}` ), and just do `const newFlagged = [...flagged]` and `newFlagged.push(imageId)` (if possible avoid abbreviations, at some point they get confusing to remember). Later when using it you could, instead of doing `flagged[src[src.length - 5]]`, you could do something like `flagged.includes(imageId)`